lib/remap-lang/src/value/object.rs

```diff
@@ -0,0 +1,327 @@
+use crate::{Object, Path, Value};
+
+impl Object for Value {
+    fn insert(&mut self, path: &Path, value: Value) -> Result<(), String> {
+        Ok(self.insert_by_path(path, value))
+    }
+
+    fn get(&self, path: &Path) -> Result<Option<Value>, String> {
+        Ok(self.get_by_path(path).cloned())
```

Comment:
Probably out of scope for this PR, but it would be nice if we could avoid this `clone` here.

As far as I'm aware the lifetime of the returned Value is always going to be the same as the Object, so it shouldn't be too hard to make it able to return a reference to the Value instead.

Reply:
Yeah, I've been thinking of building up some flame charts and checking where TRL currently spends the most time. I would love to remove a bunch of clones we do right now by starting to introduce lifetimes (we currently don't borrow any data anywhere, so there's definitely room for improvement here).